### DISCUSSION TITLE
heater SETPIONT trigger on of relay

### DISCUSSION THREAD

**Discussion by omo22** [tagyoureit/nodejs-poolController#1011] [General]
greeting to all: I have a stand alone nixie connected to Pentair VF pump RPI4, MCP23017 relay and ADS115C w10K temp all are working fine.
I am unable to figure out how to trigger auto turn on off  e.g relay#1 connected w heater fireman switch using the water 10K measured temp and selected set temp from dashboard.
relay#9 connect to valve motor diverting flow into heater
Thanks for any help!

![image](https://github.com/user-attachments/assets/6d0c4863-1e38-4125-a687-482512fbde82)
![image](https://github.com/user-attachments/assets/d34c92ca-d0ff-4072-96be-b9925098c652)

**Comment by omo22** [comment]
loging info
[replay.zip](https://github.com/user-attachments/files/16470514/replay.zip)
![image](https://github.com/user-attachments/assets/ea0aaadd-9fa2-4298-adec-4a69709d80c8)
![image](https://github.com/user-attachments/assets/e0c17c00-b5b0-44a3-a6a5-1d3699a70881)

**Comment by rstrouse** [comment]
Try setting it up as follows.

1. Remove the fireman's switch reference from the 45gpm in circuit list.  If this is simply to control the speed of the pump then add the virtual circuit Heater to the pump and when the heater comes on it will set the pump to the flow you indicated.
2. Remove the ValveHeaterOn from the circuit list.  If you want a valve to rotate when the heater comes on reference the Heater as you are in the valves setup.   However, you currently have the valve control set up to engage the fireman switch.  This is incorrect that REM reference should be to move the valve not engage the heater.
3, In the heater definition set the Gas heater to be connected to the fireman's switch.  This should be the only place that the fireman's switch relay is referenced.   When there is a call for heat, the relay will trip and turn on the heater.   Nixie will perform thermostat functions and heuristics to determine whether the heater should be on based on the setpoint and the running circuits on the controller.

**Reply by omo22** [reply]
tx so much for replay .
It works, but mainly i h had to setup heater, it must be 'min cycle time'=0 in order to test w setpoint +- triggering instant relay response. Default set is 30min.

**Comment by omo22** [comment]
one more question:
how could I link the heater : on off option to trigger a relay?
![image](https://github.com/user-attachments/assets/f2c36a07-8c5d-464b-b943-d88b2d7ca08b)

**Reply by rstrouse** [reply]
In the configuration the heater has a relay dry contact setting.
